### ISSUE TITLE
[Transform] Improve instructions for deprecated max_page_search_size transform warning

### ISSUE THREAD

**Issue by pheyos** [elastic/elasticsearch#81523]
In the 8.0 Upgrade Assistant we can see the following warning for pivot transforms using `max_page_search_size`:

![image](https://user-images.githubusercontent.com/1945390/145213023-5b855c7f-95c6-4661-ab82-ecd20ae63d65.png)

I've learned that calling the `/_transform/_upgrade` endpoint should fix this for all transforms, but this is not mentioned in the instructions (and also not mentioned in the linked docs). Instead it sounds like the user has to go to each and every pivot transform and update the settings, which can be a bit scary when there are many pivot transforms to update. I'd suggest to update the instruction message to include the recommended fix (and also let the user know that this will fix it for all transforms).

**Comment by hendrikmuhs** [comment]
I suggest to add `_upgrade` but not replacing the existing message. For this reason we can't use the same as for #81521.

#### Reasoning

From a deprecation message point of view the message is consistent with the rest of elasticsearch. The same message appears in other places, too. Currently the upgrade assistant just collects existing deprecation's. Messaging differently in the upgrade assistant compared to the other places this deprecation message appears, would be a bigger change. We somewhat have a unique feature with `_upgrade` at the moment and I agree we should advertise it more, but I think we should not remove the specific problem descriptions with a generic message.

**Comment by hendrikmuhs** [comment]
Curious about why the wording is different to the one on master I realized that PR #80144 changed the messages and reverted some changes that @lcawl and I have done, but some are also different. The messages are now out of sync between master and 7.16. This is a problem, because configurations are not the same as elasticsearch settings, we support > 7.5 transforms in 8.x, the wording must be the same and we _can't_ remove the deprecations on the master branch.

To clean up this mess:

- get master and 7.16 in sync (probably requires 2 separate PR's in parallel)
- 7.16
	- re-connect the constants from TransformDeprecations for messages and urls

## Messages

I suggest the following strings, @lcawl, @sophiec20 please review and let me know what you think (feel free to edit this comment in place).

### TransformConfig

```
- "Transform [" + id + "] is too old"
+ "Transform [" + id + "] uses an obsolete configuration format"
```

### PivotConfig
```
- "Transform [" + id + "] uses deprecated max_page_search_size"
+ "Transform [" + id + "] uses the deprecated setting [max_page_search_size]"
```
(The id must be part of the message, because this also goes into the deprecation log, it's useless if you don't know the id.)

```
- [max_page_search_size] is deprecated inside pivot. Use settings instead.
+ Remove [max_page_search_size] from the pivot and configure [max_page_search_size] in the transform settings. Alternatively use [/_transform/_upgrade] to upgrade all transforms to the latest format."
```

### TransformDeprecations

ACTION_UPGRADE_TRANSFORMS_API
```
- "Use the upgrade transforms API to fix your transforms."
+ "This transform configuration uses obsolete syntax which will be unsupported after the next upgrade. Use [/_transform/_upgrade] to upgrade all transforms to the latest format."
```

**Comment by pheyos** [comment]
@hendrikmuhs Not sure if this is relevant, but 7.16.0 is already release, so we can expect some users already generating messages with the current 7.16.0 wording.

**Comment by hendrikmuhs** [comment]
> @hendrikmuhs Not sure if this is relevant, but 7.16.0 is already released, so we can expect some users already generating messages with the current 7.16.0 wording.

I think it's fine, we can probably change messages earliest 7.17.0.

The different wordings are odd, but in the long run most will use the upgrade path: 7.x - > 7.17.y -> 8.z

**Comment by przemekwitek** [comment]
@pheyos: The messages have been changed in the PRs listed above.

After verification you can close this issue.

**Comment by pheyos** [comment]
Looks good in 7.17.0 upgrade assistant.
